python/lsst/daf/butler/core/dimensions/dataId.py

```diff
@@ -0,0 +1,351 @@
+# This file is part of daf_butler.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+__all__ = ("DataId",)
+
+import itertools
+from collections.abc import Mapping
+from .graph import DimensionGraph
+from .elements import Dimension
+
+
+class DataId(Mapping):
+    r"""A dict-like identifier for data usable across multiple collections
+    and `DatasetType`\s.
```

Comment:
To do plurals of API references, it is true that you need to use `\`. In my experience I've needed to add a space, like this (which gets eaten on render):

```rst
`DatasetType`\ s.
```

Another option, which often reads better is to never make class names plural, and instead add a plural noun. For example:

```rst
... multiple collections and `DatasetType` instances.
```

This is something I've learned from the [Google Style Guide](
https://developers.google.com/style/api-reference-comments#documentation_basics) which I'll be presenting to DM soon as an RFC.

Reply:
I had seen the form with the space as well, but found that Sphinx renders things just fine without it; does that change your opinion at all?

As for adding "instances", I think that's a good idea when it adds clarity, but adding it just to avoid a markup issue seems wrong; from the reader's perspective it's just noise-words that don't add any real information.